src/util/parented_ptr.h

```diff
@@ -26,14 +26,21 @@ class parented_ptr final {
     explicit parented_ptr(T* t) noexcept
             : m_ptr{t} {
     }
-
+// Only generate destructor if not empty, otherwise its empty but will
+// cause the parented_ptr to be not trivially destructible even though it could be.
+#ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
     ~parented_ptr() noexcept {
         DEBUG_ASSERT(!m_ptr || static_cast<const QObject*>(m_ptr)->parent());
     }
-
-    // Delete copy constructor and copy assignment operator
+#else
+    // explicitly generate trivial destructor (since decltype(m_ptr) is not a class type)
+    ~parented_ptr() noexcept = default;
+#endif
+    // Rule of 5
     parented_ptr(const parented_ptr<T>&) = delete;
     parented_ptr& operator=(const parented_ptr<T>&) = delete;
+    parented_ptr(const parented_ptr<T>&& other) = delete;
+    parented_ptr& operator=(const parented_ptr<T>&& other) = delete;
```

Comment:
I have never understood why it should be not lowed to copy or move a parented_ptr() 
A copy of the parented pointer can be safely moved around since it is not owning the pointer. It is just telling the programmer: "This is a borrowed pointer owned by the parent object".

Reply:
Ok, It makes sense to have only one instance checking the ownership. The deleted copy constructor enforces that. 
Unfortunately it prevents also to copy the object that contains it. 
A move will move the responsibility for checking the parent along with the pointer. There is nothing wrong with it. Is it?
What is your opinion here?

Reply:
I didn't really check to be honest. I just saw that we didn't follow the rule of five and thus I chose the safe default. We can lift the restrictions if we can be sure that its safe.

Reply:
I have reservations about moving. If we move an object that is owned by its parent, its only safe if the parent knows that it has been moved, do QObjects take care of telling their parent?

Reply:
Yes, it is probably OK, we can change it whenever we find a valid use case for it.